internal/runners/hello/hello_example.go

```diff
@@ -58,8 +66,40 @@ func New(p primeable) *Hello {
 	}
 }
 
-// Run contains the scope in which the hello runner logic is executed.
+// Run wraps the scope in which the hello runner logic is executed. This
+// is to ensure we catch specific errors that we are looking for and wrap
+// them in a user-facing error.
 func (h *Hello) Run(params *RunParams) error {
+	err := h.run(params)
+	if err != nil {
+		for _, unpackedError := range errs.Unpack(err) {
+			switch unpackedError.(type) {
+			case *runbits.NoNameProvidedError:
+				// Errors that we are looking for should be wrapped in a user-facing error.
+				// Ensure we wrap the top-level error returned from the runner and not
+				// the unpacked error that we are inspecting.
+				return &HelloUserFacingError{
+					locale.WrapInputError(
+						err,
+						"hello_err_no_name",
+						"Cannot say hello because no name was provided.",
+					),
+				}
+
+			default:
+				// If this is not a specific error we are looking for, do nothing.
+				continue
+			}
+		}
+
+		return locale.WrapError(err, "hello_cannot_say", "Cannot say hello.")
+	}
+
+	return nil
+}
```

Comment:
I prefer the approach that is currently in the PR. While having a function that does post-processing + `defer` fits nicely in a mental model in practice I find that when reading the code it requires more jumping around, or at least remembering that you have to jump to the defer function after reading the runner logic. This isn't a problem with the current code as you read the `Run` function, then the function that it calls, and return to the caller.

Post-processing with a `defer` also requires a named return value which is different from the _vast_ majority of the functions in our codebase and also adds another layer of complexity when reading the code.

This is all subjective, as is your suggestion, and in the end, it's your call.

Reply:
The pattern of the named return is not uncommon. It is used frequently in our codebase.

I'm open to other solutions, but between the current implementation and what I am suggesting I'd prefer my suggestion. Giving each runner a redundant `Run()` function that you have to interpret to understand does not feel like the better solution to me, but you're right, that is subjective. But, given this is meant as a reference for our codebase rather than a one-off piece of code I'm gonna be nitpicky and insist on either my solution or an alternate yet to be determined one.